docs/oracle/oracle-price-example.md

```diff
@@ -0,0 +1,247 @@
+---
+id: oracle-price-example
+title: Price Example
+---
+
+## Requirements for example
+
+You need to have a fully built oracle server. You can follow [this guide](build-oracle-server.md) to do this.
+
+You will also need a the `bitcoin-s-cli` command line tool to interact with the server.
+You can find how to build this [here](../applications/cli.md)
+
+After building the oracle server, you will need to start it with
+
+```
+./app/oracle-server/target/universal/stage/bin/bitcoin-s-oracle-server
+```
+
+## Signing BTC/USD price
+
+BTC/USD markets trade 24/7/365 around the world. Exchanges publish the market price
+for the trading pair everytime a trade is matched in their matching engine.
+
+In this example, we will
+
+1. Explain the `createdigitdecompevent` rpc
+2. Set up an oracle that can sign the BTC/USD price at a maturation time
+3. Completing the event by signing the observed market price
+
+For this example our maturation time will be
+
+>2021-02-04 00:00:00 UTC
+
+## createdigitdecompevent rpc
+
+When signing numbers, we need to use a [digit decomposition event](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Oracle.md#version-0-digit_decomposition_event_descriptor) that can sign
+each digit of a number.
+
+The rpc we need to use with bitcoin-s is `createdigitdecompevent`
+
+It takes 5 arguments
+
+1. the name for the event (`bitcoin-s-price-example`)
+2. maturation time in seconds since the epoch (`1612396800`)
+3. base (`2`)
+4. number of digits (`17`)
+5. units (`BTC/USD`)
+5. precision (idk what this is for)
+
+### Understanding digit decomposition parameters
+
+Most of these fields are self explanatory, but one that might confuse new users
+is the usage of `base` and `number of digits`. You need to set these two parameters
+in such a way that your domain is contained within `base^numdigits`.
+
+Our domain is BTC/USD price at `2020-02-04 00:00:00 UTC`. That means we need to
+pick numDigits such that `2^numDigits` can contain the BTC/USD price at `2020-02-04 00:00:00 UTC`
+
+For this example, we will pick `numDigits=17`. This means we can sign a BTC/USD price
+in between `[0,2^17-1]` or `[$0-$131,071]`. If the BTC/USD price is outside of your
+predetermined interval, you need to sign the min (`0`) or the max (`$131,071`).
+
+## Setting up the BTC/USD oracle
+
+Given the parameters we specified above, we are ready to create our digit decompisition event!
+
+```
+ ./bitcoin-s-cli createdigitdecompevent "bitcoin-s-price-example" 1612396800 2 17 "BTC/USD" 0 --rpcport 12345
```

Comment:
I don't think we should include `--rpcport 12345` then it won't work if people are using the default setup